docs/guides/implementation_blueprint/data-synchronization.md

```diff
@@ -0,0 +1,200 @@
+
+#### Creating and updating customers via the API
+
+If you want to keep your customers database up to date with Voucherify, create a customer in Voucherify every time a new user is added to your database. Use the [Create customer](ref:create-customer) API endpoint to create customer data.
+
+To update customer data, use [Update customer](ref:update-customer) API endpoint. The [Create customer](ref:create-customer) API endpoint can be also used to update customer data.
+
+<!-- Simply repeat the endpoints? -->
+
+#### Synchronizing with connectors
+
+Customer data can be upserted with integrated platforms that support outbound traffic:
+- Batch
+- Bloomreach Engagement
+- Intercom
+- Klaviyo
+- MoEngage
+- mParticle
+- Salesforce
+- Segment
+- ZOHO
+
+<!-- Again, clarification: do we want to add only those platforms that send customer 
+data to Voucherify, right? 
+
+I'll update the links once these get clarified
+
+Also: how to know which platform sends data to V% and which receives V% data?
+
+-->
+
+### Customer API endpoints
+
+<!-- TBChecked which are still supported and if this list is still up-to-date; taken from https://success.voucherify.io/article/430-crm-playbook -->
+
+The following endpoints can be used to synchronize customer data:
+
+- [List Customers](ref:list-customers)
+- [Get Customer](ref:get-customer)
+- [List Customer Activities](ref:list-customer-activities)
+- [List Customer's Segments](ref:list-customer-segments)
+- [Create Customer](ref:create-customer)
+- [Import and Update Customers using CSV](ref:import-customers-using-csv)
+- [Update Customer](ref:update-customer)
+- [Update Customer's Consents](ref:update-customers-consents)
+- [Update Customer’s consents (client-sde)](ref:update-customers-consents-client-side)
+- [Update Customers in bulk](ref:update-customers-in-bulk)
+- [Update Customers' Metadata in bulk](ref:update-customers-metadata-in-bulk)
+- [Delete Customer](ref:delete-customer)
+- [Delete Customer Permanently](ref:delete-customer-permanently)
+
+For reference, see also [the customer object](ref:customer-object) and [customer activity object](ref:customer-activity-object).
+
+<!-- do we also list all other endpoints that have a customer object that can upsert data, e.g. redeem stackable discount, check eligibility? -->
+
+## Product synchronization
+
+Products can be stored in Voucherify but it is not required. The product validation <!-- validation? --> takes place during the following actions:
+- Qualification
+- Validation
+- Redemption
+
+The product validation is based on the API payload and the data stored in Voucherify's product catalog. However, if a product does not exist in Voucherify, the validation will take into consideration the payload alone.
+
+> 🚧 Product data in the payload alone
+>
+> Unlike customer data, product data provided in the payload alone do not create a new product in Voucherify. Products need to be created or imported.
```

Comment:
There is a switch called "override" that can be added to a product or SKU. If this switch is set to true for that product, redemption will upsert the product.

Reply:
Added the following:
> In a request payload, you can add an attribute `override` with value `true` to the product object.
>
> The `override` attribute is used to enable storing the product data in Voucherify. If the product does not exist, it will be created with a `source_id`. If the product does exist, the provided values for the name, price, and metadata will replace those already stored in Voucherify.